forge-game/src/main/java/forge/game/card/Card.java

```diff
@@ -6810,6 +6810,11 @@ public List<SpellAbility> getAllPossibleAbilities(final Player player, final boo
         // this can only be called by the Human
         final List<SpellAbility> abilities = Lists.newArrayList();
         for (SpellAbility sa : getSpellAbilities()) {
+            //adventure spell check
+            if (isAdventureCard() && sa.getRootAbility().isAdventure()) {
+                if (getExiledWith() != null && CardStateName.Adventure.equals(getExiledWith().getCurrentStateName()))
+                    continue; // skip since it's already on adventure
```

Comment:
this is not what the rules say:
> 715.3d. Instead of putting a spell that was cast as an Adventure into its owner's graveyard as it resolves, its controller exiles it. For as long as that card remains exiled, that player may cast it. It can't be cast as an Adventure this way, **although other effects that allow a player to cast it may allow a player to cast it as an Adventure.**

does it still work with e. g. Ultimate of _Ashiok, Nightmare Muse_ on adventure card after this?
Otherwise I'm not sure it can be fixed without MayPlay refactor

Reply:
chould we check if the Card was exiling itself?

Reply:
> Hmn, I guess there is no generic MayPlay$ on Exile zone that could be broken by it

 Misthollow Griffin and Eternal Scourge

`You may cast CARDNAME from exile`

Reply:
I meant "cast as Adventure" + another MayPlay gives permission on it